pkg/sql/opaque.go

```diff
@@ -56,6 +58,9 @@ func buildOpaque(
 		plan, err = p.SchemaChange(ctx, stmt)
 		if err != nil {
 			return nil, err
+		} else if plan != nil {
```

Comment:
create function has a similar trick since it's not opaque and have to go through optbuilder at the moment: https://github.com/cockroachdb/cockroach/blob/f2fa926d1c73893e67314c6e951720fb46563bb5/pkg/sql/opt_exec_factory.go#L1894-L1894
Maybe better to do the counting in `p.SchemaChange` method, or I'm good with adding an extra counting there as well.

Reply:
thanks for pointing that out! i added tests for it, and moved the logic to `p.SchemaChange`

i noticed a few more places with missing telemetry, so made a followup: https://github.com/cockroachdb/cockroach/pull/105482